src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -35,17 +35,21 @@ private void ExecuteForegroundAction(Action action)
             {
                 lock (_queueGate)
                 {
-                    _foregroundTaskQueue = _foregroundTaskQueue.SafeContinueWith(
-                        _ =>
+                    _foregroundTaskQueue = _foregroundTaskQueue.SafeContinueWithFromAsync(
+                        async _ =>
                         {
+                            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.